src/index.js

```diff
@@ -1,17 +1,20 @@
 import { requestPagePreview } from './api'
+import { customEvents } from './event'
 import { createPopup } from './popup'
 import { renderPreview } from './preview'
 import '../style/index.less'
 
-function init({root, selector, lang, popupContainer}={}) {
-	const globalLang = lang || 'en'
-	root = root || document
-	selector = selector || '[data-wikipedia-preview]'
-	popupContainer = popupContainer || document.body
-
+function init({
+	root = document, 
+	selector = '[data-wikipedia-preview]', 
+	lang = 'en', 
+	popupContainer = document.body }) {
+	const globalLang = lang
 	const popup = createPopup(popupContainer)
+	const events = customEvents(popup)
+	popup.subscribe(events)
```

Comment:
Could someone please give me a high level of the relationship between `popup` and `events`, the reasoning behind subscribing to events?

Reply:
> "Preview" is the actual preview of an article. We should be able to feed it article data and unit test that is shows it correctly and it has the correct state (collapsed, expanded) based on the size of the data and that it is able to transition between those states. It should not know about the API layer.

> "Popup" is the box that shows the preview but really it doesn't know about it. It could show anything. It is responsible proper positioning. Current unit tests on this are not too bad, I think. When we do mobile (NOT IN SCOPE HERE) we can update "Popup" to handle it or, even better, create another container that acts as a bottom drawer and let the index decide which container to use depending on the platform.

"Event" is the central place where the listener binds and removes, it uses as the event connection between Popup and Preview. For example, the event subscription tells popup to close when the preview close button is clicked.